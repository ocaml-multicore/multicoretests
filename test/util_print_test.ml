type cmd = A | B | C

let show_cmd c = match c with
  | A -> "A"
  | B -> "B"
  | C -> "C"

let print_and_check_output s res =
  let ss = (String.split_on_char '\n' s) in
  List.iter (fun s -> Printf.printf "\"%s\"\n%!" s) ss;
  assert (List.equal String.equal res ss)

;;
let s1 = Util.print_triple_vertical ~fig_indent:1 show_cmd ([A],[A;B;C],[C]) in
let res1 = [
  "            |                   ";
  "            A                   ";
  "            |                   ";
  " .---------------------.";
  " |                     |                   ";
  " A                     C                   ";
  " B                                         ";
  " C                                         ";
  ""]
in
print_and_check_output s1 res1
(*assert (List.equal String.equal res1 ss1)*)
;;
let s2 = Util.print_triple_vertical ~fig_indent:1 ~center_prefix:false show_cmd ([A],[A;B;C],[C]) in
let res2 = [
  " |                                         ";
  " A                                         ";
  " |                                         ";
  " .---------------------.";
  " |                     |                   ";
  " A                     C                   ";
  " B                                         ";
  " C                                         ";
  ""]
in
print_and_check_output s2 res2
