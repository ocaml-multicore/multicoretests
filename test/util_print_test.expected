"            |                   "
"            A                   "
"            |                   "
" .---------------------."
" |                     |                   "
" A                     C                   "
" B                                         "
" C                                         "
""
" |                                         "
" A                                         "
" |                                         "
" .---------------------."
" |                     |                   "
" A                     C                   "
" B                                         "
" C                                         "
""
